R/pkg/R/SQLContext.R

```diff
@@ -189,19 +238,67 @@ createDataFrame <- function(data, schema = NULL, samplingRatio = 1.0,
           x
         }
       }
+      data[] <- lapply(data, cleanCols)
 
-      # drop factors and wrap lists
-      data <- setNames(lapply(data, cleanCols), NULL)
+      args <- list(FUN = list, SIMPLIFY = FALSE, USE.NAMES = FALSE)
+      if (arrowEnabled) {
+        shouldUseArrow <- tryCatch({
+          stopifnot(length(data) > 0)
+          dataHead <- head(data, 1)
+          # Currenty Arrow optimization does not support POSIXct and raw for now.
+          # Also, it does not support explicit float type set by users. It leads to
+          # incorrect conversion. We will fall back to the path without Arrow optimization.
+          if (any(sapply(dataHead, function(x) is(x, "POSIXct")))) {
```

Comment:
can you check -  I think `is` `is.x` doesn't something do the right thing when

head(df, 1) and one of the field is `NA`

Reply:
Looks okay in this case specifically:

```r
> any(sapply(head(data.frame(list(list(a=NA))), 1), is.raw))
[1] FALSE
> any(sapply(head(data.frame(list(list(a=NA))), 1), function(x) is(x, "POSIXct")))
[1] FALSE
> any(sapply(head(data.frame(list(list(a=1))), 1), is.raw))
[1] FALSE
> any(sapply(head(data.frame(list(list(a="a"))), 1), function(x) is(x, "POSIXct")))
[1] FALSE
> any(sapply(head(data.frame(list(list(a=raw(1)))), 1), is.raw))
[1] TRUE
> any(sapply(head(data.frame(list(list(a=as.POSIXct("2000-01-01")))), 1), function(x) is(x, "POSIXct")))
[1] TRUE
```

Reply:
LG, I tested a few cases too